pkg/provisioner/crdprovisioner.go

```diff
@@ -1217,14 +1201,33 @@ func waitForDetachVolumeFunc(obj interface{}, objectDeleted bool) (bool, error)
 	return false, nil
 }
 
-func startWatchingKube(ctx context.Context, kubeInformerFactory informers.SharedInformerFactory) {
+func (c *CrdProvisioner) AddPredicatesAndStartInformers(ctx context.Context, kubeInformerFactory informers.SharedInformerFactory, azInformerFactory azdiskinformers.SharedInformerFactory, cw *watcher.ConditionWatcher) {
```

Comment:
I don't think `AddPredicatesAndStartInformers` should be a member method of the `CrdProvisioner` as there is no direct dependency between the two. I think we can make it an independent function in one of the util or common files.